tests/framework/e2e/etcd_process.go

```diff
@@ -183,7 +180,7 @@ func (ep *EtcdServerProcess) Kill() error {
 }
 
 func (ep *EtcdServerProcess) Wait() error {
-	err := ep.proc.Wait()
+	err := ep.proc.Close()
```

Comment:
This totally changes the result. Don't know any case we would want to change it like this.

Reply:
good call, it's only used in the failpoints. Seems to work though,

Reply:
It might work, but it causes a race between previously sent SIGKILL signal and process close. Please revert.

Reply:
Reason is that you changed implementation of `Close` to always wait till the process finishes. This however not correct behavior.